lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -477,30 +481,18 @@ instance FromText ApiEpochNumber where
                 if num >= minValue && num <= maxValue then
                     Right $ ApiEpochNumber $ EpochNo $ fromIntegral num
                 else
-                    Left (err txt)
-            _ -> Left (err txt)
+                    Right $ ApiEpochNumberInvalid rest
+            _ -> Right $ ApiEpochNumberInvalid rest
       where
         minValue = fromIntegral $ minBound @Word31
         maxValue = fromIntegral $ maxBound @Word31
-        err t = TextDecodingError $ mempty
-            <> "Unable to parse epoch number: '"
-            <> T.unpack t
-            <> "'. Expecting either \"latest\" or integer from "
-            <> show minValue <> " to " <> show maxValue
 
 instance ToHttpApiData ApiEpochNumber where
     toUrlPiece = toText
 
 instance FromHttpApiData ApiEpochNumber where
     parseUrlPiece = first (T.pack . getTextDecodingError) . fromText
 
-instance ToJSON ApiEpochNumber where
-    toJSON = Aeson.String . toText
-
-instance FromJSON ApiEpochNumber where
-    parseJSON =
-        parseJSON >=> eitherToParser . bimap ShowFmt Prelude.id . fromText
```

Comment:
@Anviking @piotr-iohk  This instances are deleted as validation went downstream. In my humble opinion we do not need to resurrect roundtrips for ApiEpochNumber

Reply:
BUT, we still have roundtrip for `jsonRoundtripAndGolden $ Proxy @ApiNetworkParameters` which is very fine

Reply:
nevertheless, what Johannes proposed (in private communication) is very reasonable, ie. add textual roudtrip for `ApiEpochNumber` -> https://github.com/input-output-hk/cardano-wallet/pull/1306/commits/8d7e850f3d41a7ac784430fed4b267bc703af31f